spark/src/main/scala/io/ddf/spark/content/ViewHandler.scala

```diff
@@ -51,10 +56,50 @@ class ViewHandler(mDDF: DDF) extends io.ddf.content.ViewHandler(mDDF) with IHand
     this.get(columns, ViewFormat.withName(format))
   }
 
-  override  def getRandomSampleByNum(numSamples: Int, withReplacement: Boolean,
+  override  def getRandomSampleByNum(numSamples: Long, withReplacement: Boolean,
     seed:
   Int): DDF = {
-    null.asInstanceOf[DDF]
+
+    if (numSamples < 0) {
+      throw new IllegalArgumentException("Number of samples must be larger than or equal to 0")
+    }
+
+    if (!withReplacement){
+      mDDF.getSqlHandler.sql2ddf(s"select * from ${mDDF.getSchema.getTableName} order by rand($seed) limit $numSamples")
+    } else {
+      // First, create a Spark DF that contain random numbers from 0 to num_rows - 1
+      val numRows = mDDF.getNumRows
```

Comment:
Another way to do this is
```
rddRow.sample(true, 2.0*numSamples/rddRow.count(), seed).zipWithIndex().filter(_._2 < numSamples).map(_._1)
```
i.e. we use Spark's API to sample twice what we need, then only pick the first `numSamples` rows.
Not sure if it is more efficient though.